### PR TITLE
Create Recurrent_Neural_Network.md

### LINKED ISSUE #1115: Recurrent Neural Networks (RNN)
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/ajay-dhangar/algo/blob/main/CONTRIBUTING.md).

### Description

RNNs are a type of neural network that can process sequential data, such as text, time series, and audio. They have internal memory that allows them to retain information from previous inputs. RNNs are used for tasks like machine translation, speech recognition, and text generation.

### Self-service

- [X] I'd be willing to address this documentation request myself.

### PR DESCRIPTION
**Pull request by inkerton**
### Description
Recurrent Neural Networks (RNNs) are a class of artificial neural networks designed for modeling sequential data by introducing loops that allow information to persist across time steps. They are widely used for tasks like time-series forecasting, natural language processing (NLP), and speech recognition because of their ability to process and maintain information from previous inputs.

Fixes #1115 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/machine-learning/Recurrent_Neural_Network.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+5. **Final Prediction**:  
+   The RNN produces a final prediction, which could be a classification of the entire sequence (e.g., sentiment analysis) or a prediction for the next time step in the sequence (e.g., time-series forecasting).
+
+### Split Criteria:
+RNNs handle sequential data where each input depends on the previous elements. The network processes the data one time step at a time, updating the hidden state with each new input.
+
+### Time Complexity:
+- **Training Complexity**:  
+  The time complexity of training an RNN depends on the number of time steps \( T \), the size of the input \( n \), and the number of hidden units \( h \). It is roughly \( O(T \cdot n \cdot h^2) \).
+
```
Comment (ajay-dhangar) [inline docs/machine-learning/Recurrent_Neural_Network.md]: replace `\( T \)` with `$T$`, `\( n \)`, with `$n$`, `\( h \)` with `$h$`, and `\( O(T \cdot n \cdot h^2) \)` with `$ O(T \cdot n \cdot h^2) $`

File: docs/machine-learning/Recurrent_Neural_Network.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+- **Prediction Complexity**:  
+  For inference, the time complexity is also dependent on the sequence length, as each time step requires an update to the hidden state.
+
+### Space Complexity:
+- **Space Complexity**:  
+  The space complexity depends on the number of hidden units and time steps. Since RNNs store the hidden state at each time step, they require memory proportional to \( O(T \cdot h) \), where \( T \) is the number of time steps and \( h \) is the size of the hidden state.
+
```
Comment (ajay-dhangar) [inline docs/machine-learning/Recurrent_Neural_Network.md]: and also, replace with `$` with mathematical formula using LaTeX

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Comment by inkerton** [comment]
done

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**